src/printer.js

```diff
@@ -1698,8 +1704,13 @@ function genericPrintNoParens(path, options, print) {
         "type ",
         path.call(print, "id"),
         path.call(print, "typeParameters"),
-        " = ",
-        path.call(print, "right"),
+        " =",
+        hasLeadingOwnLineComment(options.originalText, n.right)
+          ? indent(
```

Comment:
I see. Yeah, that makes sense. Unfortunate complexity, but I'm hoping once all this works, we'll never need to touch this code again haha